### PR TITLE
Feature/fortran

### PR DESCRIPTION
**Pull request by WillTrojak**
Updated F90 Gimmik support with utilises for max line length and some changes to make future additional language support easier.

F90 extensions tested on with gcc 9.3, ifort 20200925, and nvfortran 20.9-0 w/ LLVM

### CODE REVIEW

File: gimmik/kernels/f90T.mako [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    ${dtype}, intent(out) :: c(n,${len(mat[0,:])})
+
+    integer(kind=4) :: i
+    ${dtype} :: dotp
+
+    !$omp simd
+    do i=1,n
+    % for j, jx in enumerate(mat.transpose(),start=1):
+        dotp = ${' + '.join('{kx}*b(i,{k})'.format(k=k, kx=kx)
```
Comment (FreddieWitherden) [inline gimmik/kernels/f90T.mako:18]: Can use f strings here.

File: gimmik/kernels/f90T.mako [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    ${dtype}, intent(in) :: b(n,${len(mat[:,0])})
+
+    ${dtype}, intent(out) :: c(n,${len(mat[0,:])})
+
+    integer(kind=4) :: i
+    ${dtype} :: dotp
+
+    !$omp simd
+    do i=1,n
+    % for j, jx in enumerate(mat.transpose(),start=1):
```
Comment (FreddieWitherden) [inline gimmik/kernels/f90T.mako:17]: mat.T

File: gimmik/__init__.py [hunk 3]
```diff
@@ -29,10 +52,32 @@ def generate_mm(mat, dtype, platform, alpha=1.0, beta=0.0, funcn='gimmik_mm'):
...
+
+def _line_split(maxlen,cchar,src):
+    lines = src.splitlines()
+
+    src = ''
+    for line in lines:
+        nidnt = len(line) - len(line.lstrip(' '))
+
+        while math.ceil(len(line)/maxlen) > 1:
+            ns = max(line[:maxlen].rfind('+ '),line[:maxlen].rfind('- '))
```
Comment (FreddieWitherden) [inline gimmik/__init__.py:77]: This does not seem safe to me.

File: gimmik/__init__.py [hunk 4]
```diff
@@ -1,22 +1,45 @@
...
+    cchar = {'c' : ' ',
+             'f90': '&'}
+
+    lang = pltfs[platform]
+
+    # np type to language specific types
+    try:
+        (dtype,suffix) = types[lang][dtype]
+    except KeyError:
         raise ValueError('Invalid floating point data type')
```
Comment (FreddieWitherden) [inline gimmik/__init__.py:43]: Update comment.

File: gimmik/__init__.py (hunk 3 above)
Comment (WillTrojak) [reply gimmik/__init__.py:77]: I tried some attempts to break this and it was fine, a line would never end in `- ` or `+ ` so its fine.

**Pull request closed without merging**